### PR TITLE
Dropwizard Elide Bundle and Examples

### PR DESCRIPTION
**Pull request by xiaoyao1991**
- Created `elide-contrib` submodule with `dropwizard-elide` [bundle](http://www.dropwizard.io/0.9.2/docs/manual/core.html#bundles) in
- Created `dropwizard-elide-example`

### CODE REVIEW

File: elide-example/dropwizard-elide-example/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        mysql> create database elide;
+
+3. Create ```elide``` user with password ```elide123```
+
+        mysql> grant all on elide.* to 'elide'@'localhost' identified by 'elide123';
+
+4. Launch the example webservice
+
+        ~/elide $ mvn install
+        ~/elide $ cd elide-example/elide-blog-example
```
Comment (shadanan) [inline elide-example/dropwizard-elide-example/README.md]: type

File: elide-example/dropwizard-elide-example/src/main/java/com/yahoo/elide/example/application/DropwizardElideApplication.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
+
+/**
+ * Dropwizard Application
+ */
+public class DropwizardElideApplication extends Application<DropwizardElideConfiguration> {
+    private final String name = "dropwizard-elide-example";
+
+    private final ElideBundle<DropwizardElideConfiguration> elideBundle = new ElideBundle<DropwizardElideConfiguration>(
```
Comment (shadanan) [inline elide-example/dropwizard-elide-example/src/main/java/com/yahoo/elide/example/application/DropwizardElideApplication.java]: Let's do this in a constructor – this looks weird on its own

File: elide-example/dropwizard-elide-example/src/main/java/com/yahoo/elide/example/application/DropwizardElideApplication.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        bootstrap.addBundle(elideBundle);
+    }
+
+    @Override
+    public void run(DropwizardElideConfiguration config, Environment environment) {
+        environment.jersey().register(JsonApiEndpoint.class);
+    }
+
+    public static void main(String[] args) throws Exception {
+//        new DropwizardElideApplication().run("server", "elide-example/dropwizard-elide-example/example.yml");
```
Comment (shadanan) [inline elide-example/dropwizard-elide-example/src/main/java/com/yahoo/elide/example/application/DropwizardElideApplication.java]: Print error message if args is empty

**Comment by shadanan** [comment]
:+1: after comments addressed

**Comment by yahoocla** [comment]
CLA is valid!

File: elide-contrib/dropwizard-elide/src/main/java/com/yahoo/elide/contrib/dropwizard/elide/ElideBundle.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                bind(new Elide.Builder(dataStore).auditLogger(auditLogger).build()).to(Elide.class).named("elide");
+            }
+        });
+    }
+
+    protected void configure(org.hibernate.cfg.Configuration configuration) {
+    }
+
+    @Override
+    public DataStore getDataStore(T configuration, Environment environment) {
```
Comment (DeathByTape) [inline elide-contrib/dropwizard-elide/src/main/java/com/yahoo/elide/contrib/dropwizard/elide/ElideBundle.java]: Should (could?) we generalize this to a configurable datastore? Or can we only tie this to hibernate?

Comment (DeathByTape) [inline elide-contrib/dropwizard-elide/src/main/java/com/yahoo/elide/contrib/dropwizard/elide/ElideBundle.java]: Actually, I see the interface below. It looks as though it's pretty straightforward to override what would be necessary.

**Comment by DeathByTape** [comment]
:+1:

**Pull request merged**